### PR TITLE
docs(dev): add routine maintenance steps

### LINKED ISSUE #3785: [proposal] Databend dependency management
Currently, between 800 and 1000 dependencies need to be compiled per full build/test of Databend.

**dependent-bot is not applicable**

- update one dependency at a time
- take up a lot of ci resources
- also require developer attention

## Dependency Management

1. a batch update of dependencies at the beginning of each month
    - cargo upgrade
    - cargo udeps
    - manually update dependencies that break changes
2. updates to specific dependencies as required
    - cargo audit
    - development needs

As there is currently no suitable tool to update dependencies in batch, we will perform this process manually. (The main problem is the lack of a simple way to roll back specific packages based on reported errors.)

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

docs(dev): add routine maintenance steps

## Changelog

- Documentation
- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #3785 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: docs/dev/development/routine-maintenance.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+## Upgrade the dependencies
+
+Databend involves nearly a thousand third-party dependencies from sources such as [github](https://github.com/) and [crates.io](https://crates.io/).
+
+Currently, the dependency upgrade process is divided into the following steps:
+
+- Run `cargo upgrade --workspace` to upgrade dependencies in your Cargo.toml to their latest versions.
+  - [cargo-edit](https://crates.io/crates/cargo-edit) crate needs to be installed.
+  - Please remember to perform step 4.
```
Comment (Xuanwo) [inline docs/dev/development/routine-maintenance.md:32]: At first glance, I don't know what `step 4` means, maybe it's better to use:

```
- Step 1: Run `cargo upgrade --workspace` ....
- Step 2: Check the
- Step 3: Update dependencies
- Step 4: Run `make lint` and `make test`
```

Btw, we don't need to notify readers everywhere (they just read step 1...)

We can use `**Please make sure all check passed**` in step 4 to mark it for readers.

Comment (PsiACE) [reply docs/dev/development/routine-maintenance.md:32]: got

**Pull request merged**